examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+"""
+"""
+## Introduction
+
+This example explains a "Image Enhancer/Denoiser" using UNet architecture. The problem we
+focus here is enhancing the document images which are deteriorated through external
+degradations like blur, corrupt text blocks etc... ,we have followed up on few seminal
+papers as reference and used simple pretrained UNet as encoder and Efficient Sub-Pixel
+CNN as decoder. we do not attempt super-resolution as primary task hence we return output
+image in the same dimension as input. we compute WER (Word Error Rate) metric to evaluate
+model performance helping us to assess the text information gain in real world scenarios
+when coupled with tesseract OCR engine.
+
+**References:**
+- [Enhancing OCR Accuracy with Super
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+- [Improving the Perceptual Quality of Document
+Images Using Deep Neural
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+from tensorflow.keras import callbacks
+from tensorflow.keras.applications import resnet_v2
+
+"""
+## Dataset
+we have created a synthetic document dataset that depicts several real world document
+noises, we have used blurs and morphological filters to reconstruct real world noises.
+"""
+
+"""shell
+!gdown https://drive.google.com/uc?id=1jQfHJUUQcpktQcjIzitvb4SldKwqcAMr
+!unzip -q data.zip
+"""
+
+"""
+## Dataset Preprocessing
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-80]
+train_targets = sorted(glob("data/target/*.jpg"))[:-80]
+
+val_source = sorted(glob("data/source/*.jpg"))[-80:]
+val_targets = sorted(glob("data/target/*.jpg"))[-80:]
+
+height, width = 352, 608
+batch_size = 4
+
+
+def resize_with_pad(image, height, width):
+    return tf.image.resize_with_pad(
+        image=image, target_height=height, target_width=width
+    )
+
+
+def data_preprocess(source, target):
+    source_image = tf.io.decode_png(tf.io.read_file(source), channels=3)
+    target_image = tf.io.decode_png(tf.io.read_file(target), channels=3)
+    source_image = resize_with_pad(source_image, height=height, width=width)
+    target_image = resize_with_pad(target_image, height=height, width=width)
+    source_image = resnet_v2.preprocess_input(source_image)
+    target_image = resnet_v2.preprocess_input(target_image)
```

Comment:
@anish9 
I think it's better to comment that what's the preprocess input doing here, for example commenting that input/target is normalizing to [-1, 1]. Because in the following section, to plot the samples, you denormalize the input with some static value:

```python
    noise = noise[i].numpy() + 1.0
    noise = noise * 127.5
    clean = clean[i].numpy() + 1.0
    clean = clean * 127.5
```